airflow/security/permissions.py

```diff
@@ -0,0 +1,38 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# Resource Constants
+RESOURCE_ALL_DAGS = 'AllDags'
+RESOURCE_CONFIG = 'Config'
+RESOURCE_CONNECTION = 'Connection'
+RESOURCE_DAG_CODE = 'DagCode'
+RESOURCE_DAG_PREFIX = 'DAG:'
+RESOURCE_DAG_RUN = 'DagRun'
+RESOURCE_IMPORT_ERROR = 'ImportError'
+RESOURCE_LOG = 'Log'
+RESOURCE_POOL = 'Pool'
+RESOURCE_TASK = 'Task'
+RESOURCE_TASK_INSTANCE = 'TaskInstance'
+RESOURCE_VARIABLE = "Variable"
+RESOURCE_WEBSITE = 'Website'
+RESOURCE_XCOM = 'XCom'
+
+# Action Constants
+ACTION_CAN_CREATE = 'can_create'
+ACTION_CAN_READ = 'can_read'
+ACTION_CAN_EDIT = 'can_edit'
+ACTION_CAN_DELETE = 'can_delete'
```

Comment:
I'm not sure if it'll work easily with how FAB/SQLA works, but maybe making these an enum class:


```suggestion
import enum


class Resouece(str, enum.Enum):
    ALL_DAGS = 'AllDags'
    CONFIG = 'Config'
    CONNECTION = 'Connection'
    DAG_CODE = 'DagCode'
    DAG_PREFIX = 'DAG:'
    DAG_RUN = 'DagRun'
    IMPORT_ERROR = 'ImportError'
    LOG = 'Log'
    POOL = 'Pool'
    TASK = 'Task'
    TASK_INSTANCE = 'TaskInstance'
    VARIABLE = "Variable"
    WEBSITE = 'Website'
    XCOM = 'XCom'

class Actions(str, enum.Enum):
    CAN_CREATE = 'can_create'
    CAN_READ = 'can_read'
    CAN_EDIT = 'can_edit'
    CAN_DELETE = 'can_delete'
```

What do you think?

Reply:
I by making it inherit from `str` too, which is supported by Enum, it will behave right for SQLA at least, and likely FAB too if it does anymore than pass it on.

Reply:
Lets do this later.